mne/channels/tests/test_montage.py

```diff
@@ -443,6 +450,282 @@ def test_read_dig_montage():
+        ))
+        fid.write('{rows:d} {cols:d}\n'.format(rows=N_ROWS, cols=N_COLS))
+        np.savetxt(fid, content, delimiter='\t')
+
+    return fname
+
+
+def test_read_dig_polhemus_isotrak_eeg(isotrak_eeg):
+    """Test reading Polhemus IsoTrak EEG positions."""
+    N_CHANNELS = 5
+    _SEED = 42
+    EXPECTED_FID_IN_POLHEMUS = {
+        'nasion': np.array([1.1056e-01, -5.4210e-19, 0]),
+        'lpa': np.array([-2.1075e-04, 8.0793e-02, -7.5894e-19]),
+        'rpa': np.array([2.1075e-04, -8.0793e-02, -2.8731e-18]),
+    }
+    EXPECTED_CH_POS = dict(zip(
+        ['eeg {:01d}'.format(ii) for ii in range(N_CHANNELS)],
+        np.random.RandomState(_SEED).randn(N_CHANNELS, 3),
+    ))
+
+    ch_names = [kk for kk, _ in EXPECTED_CH_POS.items()]
+    montage = read_dig_polhemus_isotrak(fname=isotrak_eeg, ch_names=ch_names)
+    assert montage.__repr__() == (
+        '<DigMontage | '
+        '0 extras (headshape), 0 HPIs, 3 fiducials, 5 channels>'
+    )
+
+    fiducials, fid_coords = _get_fid_coords(montage.dig, raise_error=False)
+
+    for kk, val in fiducials.items():
+        assert_array_equal(val, EXPECTED_FID_IN_POLHEMUS[kk])
+        assert fid_coords[kk] == FIFF.FIFFV_COORD_UNKNOWN
+
+    for kk, dig_point in zip(montage.ch_names, _get_dig_eeg(montage.dig)):
+        assert_array_equal(dig_point['r'], EXPECTED_CH_POS[kk])
+        assert dig_point['coord_frame'] == FIFF.FIFFV_COORD_UNKNOWN
+
+
+def test_read_dig_polhemus_isotrak_error_handling(isotrak_eeg, tmpdir):
+    """Test errors in reading Polhemus IsoTrak files.
+
+    1 - matching ch_names and number of points in isotrak file.
+    2 - error for unsupported file extensions.
+    """
+    # Check ch_names
+    N_CHANNELS = 5
+    EXPECTED_ERR_MSG = "not match the number of points.*Expected.*5, given 47"
+    with pytest.raises(ValueError, match=EXPECTED_ERR_MSG):
+        _ = read_dig_polhemus_isotrak(
+            fname=isotrak_eeg,
+            ch_names=['eeg {:01d}'.format(ii) for ii in range(N_CHANNELS + 42)]
+        )
+
+    # Check fname extensions
+    fname = op.join(tmpdir, 'foo.bar')
+    with pytest.raises(
+        ValueError,
+        match="Allowed val.*'.hsp', '.elp' and '.eeg', but got '.bar' instead"
+    ):
+        _ = read_dig_polhemus_isotrak(fname=fname, ch_names=None)
+
+
+def test_combining_digmontage_objects():
+    """Test combining different DigMontage objects."""
+    rng = np.random.RandomState(0)
+    fiducials = dict(zip(('nasion', 'lpa', 'rpa'), rng.rand(3, 3)))
+
+    # hsp positions are [1X, 1X, 1X]
+    hsp1 = make_dig_montage(**fiducials, hsp=np.full((2, 3), 11))
+    hsp2 = make_dig_montage(**fiducials, hsp=np.full((2, 3), 12))
+    hsp3 = make_dig_montage(**fiducials, hsp=np.full((2, 3), 13))
+
+    # hpi positions are [2X, 2X, 2X]
+    hpi1 = make_dig_montage(**fiducials, hpi=np.full((2, 3), 21))
+    hpi2 = make_dig_montage(**fiducials, hpi=np.full((2, 3), 22))
+    hpi3 = make_dig_montage(**fiducials, hpi=np.full((2, 3), 23))
```

Comment:
I think these were the problem on 3.5, passing `23.` should fix it (pushed a commit)